tests/testthat/test-select-helpers.R

```
@@ -29,6 +29,18 @@ test_that("throws with empty pattern is provided", {
   expect_error(matches(""))
 })
 
+test_that("works though passed variable's name matches one of the var names (issue #2184)", {
+  skip("Currently failing")
+  vars <- "x"
+  names(vars) <- vars
+
+  expect_equal(select_vars(vars, starts_with(x)), c(x = "x"))
```

I think this test should throw an error, because the variable x isn't defined. Did you mean to ask `starts_with(vars)` ?